system/jhtree/keybuild.cpp

```diff
@@ -201,8 +204,13 @@ class CKeyBuilderBase : public CInterface
         if (out)
         {
             out->flush();
-            out->seek(0, IFSbegin);
-            keyHdr->write(out, crc);
+            if (keyHdr->getKeyType() & USE_TRAILING_HEADER)
```

Comment:
Is there a problem with updating the crcs? 
This may need to update a different crc variable. since it is the crc of an extra block, rather than the header that is being written.